src/Model/APContact.php

```diff
@@ -291,14 +291,11 @@ public static function getByURL(string $url, $update = null): array
 			return $fetched_contact;
 		}
 
-		$parts = parse_url($apcontact['url']);
-		unset($parts['scheme']);
-		unset($parts['path']);
-
 		if (empty($apcontact['addr'])) {
-			if (!empty($apcontact['nick']) && is_array($parts)) {
-				$apcontact['addr'] = $apcontact['nick'] . '@' . str_replace('//', '', Network::unparseURL($parts));
-			} else {
+			try {
+				$apcontact['addr'] = $apcontact['nick'] . '@' . (new Uri($apcontact['url']))->getAuthority();
```

Comment:
What exactly is `getAuthority` returning? The replaced method took the port number into account. So server at port 10000 had been: `user@domain.tld:10000`.

Reply:
And this would be what the original snippet would have returned as well since it was only stripping the scheme and the path. But I'm glad you were able to know exactly what is returned now thanks to the use of the UriInterface object.

Reply:
Well, with the return value of `getAuthority` I'm slightly helpless, since the name of the function doesn't ring a bell. But a look in the replaced coding told be what was expected. So this other function is doing the same?

Reply:
URL authority is `[user[:password]@]domain.tld[:port]`.